cli_ftd/user/sensor/stm32/stm32_i2c.h

```diff
@@ -21,10 +23,24 @@ uint32_t stm32__time_of_flight(void);
 void stm32__accelerometer(int32_t arr[]);
 void stm32__gyroscope(int32_t arr[]);
 void stm32__magnetometer(int32_t arr[]);
+void stm32__heartbeat_data(uint16_t arr[]);
+void stm32__gps_data(float data_arr[]);
 
 // Helper functions
 bool stm32__write_data(uint8_t write_address, uint8_t data);
 bool stm32__read_data(uint8_t write_address, uint8_t readBuf[],
                       uint32_t readCount);
 
+uint32_t stm32__convert_to_uint32(uint8_t *data);
```

Comment:
Please push these into the utility headers
This module should only contain the i2c slave transaction APIs